# train
batch_size: &batch_size 64
epochs: 120
start_epoch: 0
launcher: 'none'
ckpt_save_interval: 1
max_ckpt_save_num: 20

# global settings
point_cloud_range: &point_cloud_range [-32.0, -16.0, -2.5, 32.0, 32.0, 1.5]
voxel_size: &voxel_size [0.16, 0.16, 4]
num_points_per_trajectory: &num_points_per_trajectory 10 # ~ 20m length

# dataset settings
dataset_type: &dataset_type "MTPDataset"
data_root: &data_root "/workspace/data/KITTI_RAW/trajectory_prediction/"
num_modes: &num_modes 3 # number of trajectory prediction modalities

data:  
  train:
    type: *dataset_type
    root_path: *data_root
    mode: 'train'
    subset: ['00', '02', '05', '07']
    use_cache: true
    use_lidar_bev: true
    use_lidar_points: false
    lidar_intensity_max: 1.0
    point_cloud_range: *point_cloud_range
    num_points_per_trajectory: *num_points_per_trajectory
    voxel_size: *voxel_size
    batch_size: *batch_size
    num_workers: 4
    road_width: 2.0
    traj_thresh: 2.0
    num_modes: *num_modes
 
  val:
    type: *dataset_type
    root_path: *data_root
    mode: 'val'
    subset: ['10']
    use_cache: true
    use_lidar_bev: true
    use_lidar_points: false
    lidar_intensity_max: 1.0
    point_cloud_range: *point_cloud_range
    num_points_per_trajectory: *num_points_per_trajectory
    voxel_size: *voxel_size
    batch_size: *batch_size
    num_workers: 4
    road_width: 2.0
    num_modes: *num_modes
    
# model settings
model:
  type: PIXORPredictor
  task: 'trajectory_prediction'

  backbone:
    use_bn: true
    input_dim: 27
    num_blocks: [3, 6, 6, 3]

  header:
    use_bn: true
    weight_loss_seg: 1.0 
    mtp:
      anchor_path: 'data/kitti/'
      with_anchor: true
      with_hist_traj: false
      num_modes: *num_modes
      input_dim: 1024
      n_hidden_layers: 4096
      num_points_per_trajectory: *num_points_per_trajectory
